src/model/compressedhistory.js

```diff
@@ -0,0 +1,150 @@
+/**
+ * @license Copyright (c) 2003-2016, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+'use strict';
+
+import History from './history.js';
+
+/**
+ * `CompressedHistory` keeps deltas applied to the {@link engine.model.Document document} in their transformed state. Deltas
+ * stored in `CompressedHistory` might get updated, split into more deltas or removed as other deltas are applied to the document.
+ * Modifying the original deltas history results in a compressed version of history and makes some scripts faster and easier to implement.
+ *
+ * **Note:** deltas kept in `CompressedHistory` should be used only to transform deltas. Do not use `CompressedHistory` to get original
+ * delta (especially basing on its {@link engine.model.delta.Delta#baseVersion baseVersion}). Do not trust base versions of deltas
+ * returned by `CompressedHistory`. After transforming your delta by deltas from `CompressedHistory`, fix its base version accordingly.
+ *
+ * @see engine.model.History
+ * @memberOf engine.model
+ */
+export default class CompressedHistory extends History {
```

Comment:
`CompressedHistory` has nothing to do with compression. Also I do not see any reason to keep `CompressedHistory` and `History` separated.